src/main/scala/nl.knaw.dans.easy.solr4files/components/Bag.scala

```diff
@@ -17,31 +17,37 @@ package nl.knaw.dans.easy.solr4files.components
 
 import java.net.URL
 
-import nl.knaw.dans.easy.solr4files.{ FileToShaMap, SolrLiterals }
+import nl.knaw.dans.easy.solr4files.{ FileToShaMap, SolrLiterals, _ }
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 
 import scala.util.Try
 import scala.xml.Elem
 
 case class Bag(storeName: String,
                bagId: String,
-               private val vaultIO: VaultIO
+               private val vault: Vault
               ) extends DebugEnhancedLogging {
 
-  private def getDepositor: String = Try {
+  private def getDepositor: String = {
     val key = "EASY-User-Account"
-    vaultIO.linesFrom(storeName, bagId, "bag-info.txt")
-      .filter(_.trim.startsWith(key))
-      .map(_.trim.replace(key, "").trim.replace(":", "").trim)
-      .head
-  }.getOrElse("unknown")
+    for {
+      url <- Try(vault.fileURL(storeName, bagId, "bag-info.txt"))
+      lines <- url.readLines
+      wantedLines = lines.filter(_.trim.startsWith(key))
+      values = wantedLines.map(_.trim.replace(key, "").trim.replace(":", "").trim)
+      value <- Try(values.head)
```

Comment:
I keep repeating this point: you can't do a `head`! It's dangerous, even though you wrapped it in a `Try` (which in itself isn't necessary here either). What happens if `values` is empty?

Instead either do a pattern match, or in this case even better: use the failsafe `headOption`. Then you have to deal with the option, but with that you can specify a custom exception...